src/Convert.cpp

```diff
@@ -62,12 +62,12 @@ QString hexValue(int value, int width)
 	                      
 }
 
-QString& escapeXML(QString& str)
+QString escapeXML(const QString& str)
 {
-	return str.replace('&', "&amp;").replace('<', "&lt;").replace('>', "&gt;");
+	return QString(str).replace('&', "&amp;").replace('<', "&lt;").replace('>', "&gt;");
```

Comment:
Detail, small optimization.

`{un}escapeXML()` are two functions that unconditionally make a copy of the `str` parameter. In cases like this it's often better (for performance) to take this parameter by-value instead of by-const-reference.

The difference is that:
* Currently the function always makes a copy. (And the caller passes a reference).
* In the pass-by-value case, the caller must create the `str` object. Either by making a copy, or by moving an r-value. So possibly a (more expensive) copy is changed into a (less expensive) move operation.

For example, later on in this patch you have the sub-expression:
```    unescapeXML(bp.mid(p, q - p).simplified())```

The `simplified()` method returns a `QString`, this is an r-value (it's an unnamed temporary). Currently that temporary is passed by reference and then `unescapeXML()` makes a copy. If instead this parameter is passed by-value, then this temporary is moved into `unescapeXML()`, so no copy at all.

In other cases the parameter passed to `{un}escapeXML()` might be an l-value. Then the compile cannot move and must make a copy. But because the function itself doesn't make a copy anymore, this is still only one copy. The difference is whether the caller or the callee makes the copy.

Reply:
To be thorough, shouldn't we consider creating two versions of each function then? One for normal lvalue references and another for rvalue references? Something like:

```
QString escapeXML(const QString& str)
{
    return QString(str).replace('&', "&amp;").replace('<', "&lt;").replace('>', "&gt;");
}

QString escapeXML(QString&& str)
{
    return QString(std::move(str)).replace('&', "&amp;").replace('<', "&lt;").replace('>', "&gt;");
}
```